lib/opensrs/server.rb

```diff
@@ -75,13 +87,29 @@ def http
       http
     end
 
+    def sanitize(type, data, options)
+      return data unless @sanitize_logs
+
+      SANITIZING_METHODS.inject(data) do |current_data, method|
+        send(method, type, current_data, options)
+      end
+    end
+
+    def sanitize_rw_register(type, data, options)
+      return data unless type == "Request" &&
+                         options[:object] == "DOMAIN" &&
+                         options[:action] == "SW_REGISTER"
+
+      data.gsub(%r{(<item key="reg_password">).*(</item>)}, '\1**sanitized**\2')
```

Comment:
@jsaubry Is there no way you can use the XML processor to do this rather than using regular expressions?

Reply:
I'll take a look into that

Reply:
It would be possible, but the logging would be done after the the parsing and wouldn't be complete since the parser drop some part of the responses. Currently,
```ruby
begin
  response = http.post(server_path, xml, headers(xml))
  log('Response', response.body, data)
  rescue Net::HTTPBadResponse
    raise (...)
  end
  parsed_response = xml_processor.parse(response.body)
```

This would change the current behavior of the logger. Do you prefer that over using the regex?